Find a functionally equivalent code:android.hardware.input.InputManager.InputDeviceVibrator.vibrate:COMMENT
Method Modifier: public      hidden      
Comment:/**
 * @hide
 */

@@ -1,3 +1,10 @@
 {
-    vibrate(pattern, repeat);
+    if (repeat >= pattern.length) {
+        throw new ArrayIndexOutOfBoundsException();
+    }
+    try {
+        mIm.vibrate(mDeviceId, pattern, repeat, mToken);
+    } catch (RemoteException ex) {
+        Log.w(TAG, "Failed to vibrate.", ex);
+    }
 }

